cloudant-sync-datastore-core/src/test/java/com/cloudant/common/UnreliableProxyTestBase.java

```diff
@@ -0,0 +1,118 @@
+import eu.rekawek.toxiproxy.ToxiproxyClient;
+import eu.rekawek.toxiproxy.model.Toxic;
+import eu.rekawek.toxiproxy.model.ToxicDirection;
+
+import java.util.UUID;
+
+/**
+ * Created by tomblench on 23/07/15.
+ */
+public class UnreliableProxyTestBase extends ReplicationTestBase {
+
+    // Proxy configuration
+    private final static String PROXY_HOST = System.getProperty("test.couch.proxy.host",
+            "localhost");
+    // this is the toxiproxy admin port for setting up proxies and toxics
+    private final static int PROXY_ADMIN_PORT = Integer.parseInt(System.getProperty("test.couch" +
+                    ".proxy.admin.port",
+            "8474"));
+    // this is where the toxiproxy will listen (note uses test.couch.port so the tests will use the
+    // proxy not the real couch port).
+    private final static int PROXY_PORT = Integer.parseInt(System.getProperty("test.couch.port",
+            "8000"));
+
+    // Couch configuration
+    private final static String COUCH_HOST = System.getProperty("test.couch.host", "localhost");
+    // this is where couchdb listens
+    private final static int COUCH_TARGET_PORT = Integer.parseInt(System.getProperty("test.couch" +
+                    ".proxy.target.port",
+            "5984"));
+
+    // A toxiproxy client to administer the toxiproxy config
+    private static ToxiproxyClient toxicClient = new ToxiproxyClient(PROXY_HOST, PROXY_ADMIN_PORT);
+
+    // A toxiproxy instance for running unreliability tests
+    private static Proxy toxiProxy = null;
+
+    /**
+     * Set up a proxy with a unique name for this test run, all requests will be routed through the
+     * proxy, but it will only be toxic after some toxics are added to it by the tests.
+     *
+     * @throws Exception
+     */
+    @BeforeClass
+    public static void addProxy() throws Exception {
+        // For now we proxy all tests in the same way
+        toxiProxy = toxicClient.createProxy(UUID.randomUUID().toString(), PROXY_HOST + ":" +
+                PROXY_PORT, COUCH_HOST + ":" + COUCH_TARGET_PORT);
+    }
+
+    /**
+     * Remove any added toxics after each test
+     *
+     * @throws Exception
+     */
+    @After
+    public void removeToxics() throws Exception {
+        for (Toxic t : toxiProxy.toxics().getAll()) {
+            t.remove();
+        }
+    }
+
+    /**
+     * Delete the proxy created by this test run
+     *
+     * @throws Exception
+     */
+    @AfterClass
+    public static void deleteProxy() throws Exception {
+        if (toxiProxy != null) {
+            toxiProxy.delete();
+        }
+    }
+
+    /**
+     * Name for the default timeout toxic
+     */
+    private static final String TIMEOUT_NAME = "Timeout_50ms_0.5";
+
+    /**
+     * Add a default timeout toxic
+     *
+     * @throws Exception
+     */
+    protected void addTimeoutToxic() throws Exception {
+        toxiProxy.toxics().timeout(TIMEOUT_NAME, ToxicDirection.DOWNSTREAM, 50).setToxicity
```

Comment:
I think the timeout and toxicity values might be better pulled out into constants and placed with the `TIMEOUT_NAME`. It might even be worth constructing `TIMEOUT_NAME` from the constants so it stays in sync with the values if we change them.